src/crimson/os/seastore/cache.h

```diff
@@ -562,11 +560,17 @@ class Cache {
 
   using src_ext_t = std::pair<Transaction::src_t, extent_types_t>;
   struct {
-    std::unordered_map<Transaction::src_t, uint64_t> trans_created;
-    std::unordered_map<Transaction::src_t, uint64_t> trans_committed;
+    std::vector<uint64_t> trans_created_by_src;
```

Comment:
after a second thought, `std::array<uint64_t, static_cast<size_t>(Transaction::src_t::MAX)>` would be more appropriate. as the sizes of these containers are known at compile time.

Reply:
I think size consideration is not a big matter here as Cache and its stats are singleton. And we already resize the vector to a proper size during initialization ...

Reply:
agreed. it's not a big deal from the performance perspective. it's just "more appropriate and simpler". and i don't feel strong either way.

FWIW, resizing vector does not change its capacity in some cases, see http://en.cppreference.com/w/cpp/container/vector/resize#Notes

Reply:
I'd vote with Kefu on that: we should use std::array where possible.

Reply:
see https://github.com/ceph/ceph/pull/42202/commits/8a5bbd2a79db43cf41d012ad7260894f4d933c75